### ISSUE TITLE
Argo Workflow EventSource

### ISSUE THREAD

**Issue by keshamin** [argoproj/argo-events#1350]
**Is your feature request related to a problem? Please describe.**
I want to set up pretty basic scenario: one workflow starts after another workflow succeeds. I want the workflows to be decoupled: I don't want one workflow to explicitly submit another. It should be implemented via events.

I wanted to use Resource EventSource for that, but if I subscribe of `v1/Event` with reason `WorkflowSucceed` (or something like that) the event payload doesn't contain any metadata to filter on. Like labels/annotations. It just have ID of involvedObject. So I cannot subscribe on successful runs of some `CronWorkflow`.

If I subscribe on UPDATES of Workflow object with filter by label `workflows.argoproj.io/phase: Succeeded` I don't have any guarantee that object will not change after that. If it does, that would produce redundant undesired triggers. 

So basically, the only solution I have now is to create a special webhook and invoke it in the end of origin workflow to register an event. But it is pretty ugly solution, considering that Argo Events and Argo Workflows are parts of the same ecosystem. 

**Describe the solution you'd like**
I'd like to have a special event source that would allow to subscribe on events, related to Argo Workflows.

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
There are a lot more information in the k8s events can be used for field filtering, `kubect get events -o yaml`

**Comment by ciiiii** [comment]
@keshamin Maybe this feature https://github.com/argoproj/argo-events/issues/1445 can resolve your problem.

**Comment by AndreLobato** [comment]
I am surprised this has not come up more often, it seems such a natural solution. What prevent us to have an WorkflowStatus EventSource for example, that listen for any status change from the mentioned workflows. Couldn't maybe be hooked into the same way the UI do?
